bindings/go/input/helm/blob.go

```diff
@@ -30,13 +31,12 @@ var (
 	ErrUnsupportedField = errors.New("unsupported input field must not be used")
 )
 
-// ReadOnlyChart contains Helm chart contents as tgz archive and some metadata, required for packaging into OCI Layout.
+// ReadOnlyChart contains Helm chart contents as tgz archive, some metadata and optionally a provenance file.
 type ReadOnlyChart struct {
-	Name    string
-	Version string
-	Size    int64
-	Digest  digest.Digest
-	Content io.ReadCloser
+	Name      string
+	Version   string
+	ChartBlob *filesystem.Blob
```

Comment:
arguably this should be a blob.ReadOnlyBlob with needed assertions in the code imo

Reply:
Now that I think about it, this is fine for now. :)